raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -144,4 +172,14 @@ else if (!that.endOffset.isPresent())
         }
     }
 
+    private static final Comparator<ReplicaState> FETCH_TIMESTAMP_COMPARATOR = (state, that) -> {
+        if (state.lastFetchTimestamp.equals(that.lastFetchTimestamp))
```

Comment:
I think you compare the nodeId to produce some deterministic ordering, but technically it's not necessary right?

Reply:
Yup it is not, but I guess it's no harm too :)